### PR TITLE
ci/openshift-ci: Move openshift-ci from the tests repo here

### PR DESCRIPTION
**Pull request by ldoktor**
The https://github.com/kata-containers/tests repo should become read-only soon, let's move the openshift-ci scripts here and modify them to work well from within this repository

Fixes: #8653

### CODE REVIEW

**Comment by wainersm** [comment]
@ldoktor I recommend splitting up the webhook pieces in another PR.

This webhook was created in order to run the Kubernetes e2e tests in our CI by adding the `runtimeClass` in the spec of any pod created. Technically this webhook can be used by end users to transform pod specs, e.g., as @liubin suggested another webhook in https://github.com/kata-containers/kata-containers/issues/3976 . So I believe opening a PR specifically to discuss the fate of the webhook in kata-containers/kata-containers would bring more attention from the broaden community to this topic.

**Comment by ldoktor** [comment]
> @ldoktor I recommend splitting up the webhook pieces in another PR.
> 
> This webhook was created in order to run the Kubernetes e2e tests in our CI by adding the `runtimeClass` in the spec of any pod created. Technically this webhook can be used by end users to transform pod specs, e.g., as @liubin suggested another webhook in #3976 . So I believe opening a PR specifically to discuss the fate of the webhook in kata-containers/kata-containers would bring more attention from the broaden community to this topic.

I don't have any preferences with regards to webhook implementation, as long as it works I'm fine with it. That #3976 is from March 2022, is it likely to be polished and merged by the end of this year to be ready for the tests repo deprecation? Otherwise I'd rather bring what is already existing here and remove that once the #3976 is delivered.

File: tools/testing/lib/common.bash
Comment (GabyCT) [inline tools/testing/lib/common.bash:1]: This file already exists in the kata containers repository, this is a duplication

File: tools/testing/lib/common.bash [hunk 1]
```diff
@@ -0,0 +1,376 @@
+#!/usr/bin/env bash
+#
```
Comment (GabyCT) [inline tools/testing/lib/common.bash:2]: https://github.com/kata-containers/kata-containers/blob/main/tests/common.bash

File: tools/testing/lib/common.bash
Comment (ldoktor) [reply tools/testing/lib/common.bash:1]: Ah, you are right. It's not identical but should do. I'll test things a bit more and update probably tomorrow.

**Comment by ldoktor** [comment]
Removed the `tools/testing/lib` as very similar one is present in `tests` and pointed to that one in the pipeline scripts. No other changes.

File: ci/openshift-ci/lib.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    export GOPATH=${GOPATH:-$(go env GOPATH)}
+else
+    # if go isn't installed, set default location for GOPATH
+    export GOPATH="${GOPATH:-$HOME/go}"
+fi
+
+lib_dir=$(dirname "${BASH_SOURCE[0]}")
+source "$lib_dir/../../tests/common.bash"
+
+export katacontainers_repo=${katacontainers_repo:="github.com/kata-containers/kata-containers"}
```
Comment (wainersm) [inline ci/openshift-ci/lib.sh:19]: Just curious: ':=' is equivalent to ':-'. Any reason to opt for the former?

**Comment by wainersm** [comment]
hi @ldoktor ! As we talked in private, let's try to merge https://github.com/kata-containers/tests/pull/5799 first then this PR rebased. So I won't trigger the CI of this PR to save resources, but it LGTM.

**Comment by wainersm** [comment]
> hi @ldoktor ! As we talked in private, let's try to merge [kata-containers/tests#5799](https://github.com/kata-containers/tests/pull/5799) first then this PR rebased. So I won't trigger the CI of this PR to save resources, but it LGTM.

https://github.com/kata-containers/tests/pull/5799 is merged now. @ldoktor could you please rebase?

**Comment by ldoktor** [comment]
Changes:

* Edited the first commit to use f15be37d9bef58a0128bcba006f8abb3ea13e8da version files from tests repo (second commit applied cleanly, tested on kubernetes)

**Comment by ldoktor** [comment]
> Changes:
> 
>     * Edited the first commit to use f15be37d9bef58a0128bcba006f8abb3ea13e8da version files from tests repo (second commit applied cleanly, tested on kubernetes)

Note for those who'd like to re-try, the current version fails when the node does not support selinux. I do mean to address that once we move everything here but for testing purposes you can just execute `./test.sh` which fails to apply the selinux hack. But as that is the last step everything is prepared so you can just run `./run_smoke_test.sh` to ensure everything is set correctly. Obviously running on OCP with RHCOS works well.

**Comment by wainersm** [comment]
hi @ldoktor !

I'd like to fix all static checks errors before triggering the CI jobs.

`Static checks / static-checks (make static-checks) (pull_request) ` failed because `tools/testing/kata-webhook/README.md` is a "dangling README`. The project requires all README files be linked all way down to the README in the root directory. I think if you add an entry in https://github.com/kata-containers/kata-containers/blob/main/README.md#additional-components the checker will pass.

**Comment by ldoktor** [comment]
Changes:

* New commit to link the webhook's README from the main README file

**Review by wainersm — APPROVED** [review]

**Comment by ldoktor** [comment]
Changes:

* Clean rebase, no code changes

**Comment by wainersm** [comment]
hi @GabyCT ! Can we get your approval on this one as well?

**Review by GabyCT — APPROVED** [review]

**Comment by GabyCT** [comment]
@wainersm I just did it :)

**Pull request merged**